clients/segments/segments.go

```diff
@@ -178,14 +164,14 @@ func (c Client) GetAll(ctx context.Context) ([]api.Response, error) {
 		Uid string `json:"uid"`
 	}
 	if err = json.Unmarshal(listResp.Data, &segments); err != nil {
-		return nil, fmt.Errorf(errMsg, "get all", err)
+		return nil, intErr.ErrorRuntime{Resource: resource, Reason: "unmarshalling failed", Wrapped: err}
```

Comment:
The `ErrorRuntime` error is always used when an error occurs, right? Would the reason here not be duplicate information compared to the returned error?